corehq/apps/hqadmin/utils.py

```diff
@@ -175,3 +178,19 @@ def unset_password(user):
     # 128 bits / 8 (bits/byte) = 16 bytes
     random_key = os.urandom(16).hex()
     user.set_password(random_key)
+
+
+def get_download_url(content, name, content_type=None, timeout=24 * 60):
+    """Upload file to blob storage for subsequent download"""
+    unique_id = str(uuid.uuid4())
+    get_blob_db().put(
+        content,
+        domain=unique_id,
+        parent_id=unique_id,
+        type_code=CODES.tempfile,
+        key=unique_id,
+        name=name,
+        content_type=content_type,  # optional
+        timeout=timeout,  # minutes
+    )
```

Comment:
@millerdev I'm pinging you on this PR first because you have been involved in discussions about this issue, but especially because I'm not confident that my use of the blobdb is entirely appropriate.

First off, I wasn't sure what a reasonable value for `domain` or `parent_id` would be in this context, so I just used the key.
Secondly, I'm re-using `CODES.tempfile`, since that's basically what this is, but I'd be happy to add a new type code if that seems more appropriate.

Reply:
Oh yeah that could work too - I did think about trying to do a technically valid but not actually possible domain name, like something uppercase, for instance, but didn't want to risk getting in the way of future constraints being added to that field.  That's perhaps not a realistic concern, though.

Reply:
Oh nice, I'll update that, thanks